### PR TITLE
add tests for validation module

### PR DESCRIPTION
**Pull request by AurelienJaquier**
also handle edge case where efel features can be `None` in `hyperpolarization_test`.

### CODE REVIEW

File: tests/test_validation/test_validation.py [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+    assert result["passed"] is True
+    assert result["name"] == "Simulatable Neuron Hyperpolarization Validation"
+    assert (
+        result["validation_details"]
+        == "Validation passed: Hyperpolarized voltage (-1.00 mV) is lower than RMP (0.00 mV)."
+    )
+    assert len(result["figures"]) == 1
+    assert result["figures"][0] == dummy_out_dir / "hyper.pdf"
+
+    # hyperpolarized voltage < RMP case
```
Comment (ilkilic) [inline tests/test_validation/test_validation.py:264]: I think you meant > instead of < here, the neuron didn’t hyperpolarize in this case.

Comment (AurelienJaquier) [reply tests/test_validation/test_validation.py]: Good catch. Fixed in latest commit
